packages/core/src/scripts/generate-metadata.ts

```diff
@@ -0,0 +1,399 @@
+  const fileName = path.basename(filePath, path.extname(filePath));
+
+  // Only look for sub-components if we're in a component directory
+  if (!dirName.match(/^[A-Z]/)) return [];
+
+  // Only include sub-components for the main component (the one that matches the directory name)
+  if (fileName !== dirName) return [];
+
+  return allFiles
+    .filter(f => {
+      const fDir = path.dirname(f);
+      const fName = path.basename(f, path.extname(f));
+
+      // Only include files that:
+      // 1. Are in the same directory
+      // 2. Start with the parent component name (e.g., "TipseenContent" for "Tipseen")
+      // 3. Are not the parent component itself
+      return fDir === dir && fName !== dirName && fName.startsWith(dirName);
+    })
+    .map(f => path.basename(f, path.extname(f)));
+}
+
+/**
+ * Merges aggregator records with docgen results and flattens the structure
+ */
+function mergeResults(aggregator: AggregatorRecord[], docgen: DocgenResult[]): FinalOutput {
+  const docgenMap = new Map<string, DocgenResult>();
+  for (const d of docgen) {
+    docgenMap.set(d.filePath, d);
+  }
+
+  // Get all file paths for sub-components lookup
+  const allFilePaths = aggregator.map(agg => agg.filePath);
+
+  const files = aggregator.flatMap(agg => {
+    const doc = docgenMap.get(agg.filePath);
+    if (!doc) return [];
+
+    return doc.components.map(component => ({
+      filePath: toRelativePath(agg.filePath),
+      aggregator: agg.aggregator,
+      symbols: agg.symbols,
+      displayName: component.displayName,
+      description: component.description,
+      props: component.props,
+      import: `import { ${component.displayName} } from "@vibe/core${agg.aggregator === "next" ? "/next" : ""}"`,
+      parentComponent: getParentComponent(toRelativePath(agg.filePath)),
+      subComponents: findSubComponents(toRelativePath(agg.filePath), allFilePaths.map(toRelativePath))
+    }));
+  });
+
+  return files;
+}
+
+/**
+ * Unifies type definitions with their corresponding component files
+ */
+function unifyTypesWithComponents(records: AggregatorRecord[]): AggregatorRecord[] {
+  const newRecords: AggregatorRecord[] = [];
+  const pathMap = new Map<string, AggregatorRecord>();
+  for (const r of records) {
+    pathMap.set(r.filePath, r);
+  }
+
+  for (const r of records) {
+    if (r.filePath.endsWith(".types.ts")) {
+      const baseName = path.basename(r.filePath, ".types.ts");
+      const dirName = path.dirname(r.filePath);
+      const candidateTsx = path.join(dirName, `${baseName}.tsx`);
+      const candidateTs = path.join(dirName, `${baseName}.ts`);
+
+      const mainRecord = pathMap.get(candidateTsx) || pathMap.get(candidateTs);
+      if (mainRecord) {
+        mainRecord.symbols = Array.from(new Set([...mainRecord.symbols, ...r.symbols]));
+      }
+      continue;
+    }
+
+    newRecords.push(r);
+  }
+
+  return newRecords;
+}
+
+function main() {
+  const startTime = performance.now();
```

Comment:
Do we need it in the CI?

Reply:
Just for debugging, I've added it intentionally for CI